### DISCUSSION TITLE
Add support for FedCM (Federated Credential Management API)

### DISCUSSION THREAD

**Discussion by tbroyer** [keycloak/keycloak#20339] [Ideas]
The phase-out of third-party cookies will break OIDC Front-channel logout and OIDC Session management, as well as the _silent `check-sso`_ and Session Status iframe of the Keycloak JavaScript adapter.

Proposed replacements are [First-Party Sets (FPS)](https://developer.chrome.com/docs/privacy-sandbox/first-party-sets/) and [FedCM](https://fedidcg.github.io/FedCM/). While Keycloak could probably help with FPS by supporting a `/.well-known/first-party-set.json`, I'm more interested in FedCM support here (if anyone's interested in FPS, please open a separate feature request).

To support FedCM, AFAICT (see also [Google's Developer Guide](https://developer.chrome.com/docs/privacy-sandbox/fedcm/)), Keycloak would have to:
* expose a `/.well-known/web-identity` (if not deployed at an eTLD+1, this could at least help admins configure the eTLD+1 to return the same content, possibly just using a a reverse proxy for that file); the file would list all realms (possibly filtered if that could be enabled/disabled per-realm)
* expose an IdP config file for each realm; the `branding.icons` could possibly come from the ["Logo URL" in the Advanced tab](https://www.keycloak.org/docs/latest/server_admin/index.html#con-advanced-settings_server_administration_guide), other `branding` options would have to be configurable (those are optional, so the first implementation might not need to support that, and just omit those fields)
* implement an accounts list endpoint (all the information is already here, and should only be presented in the expected JSON format; note that the account ID is not shared with the RP so pairwise subs are still possible)
* implement a client metadata endpoint (the information is already here in the "Advanced" tab)
* implement an ID assertion endpoint (this will generate an access token for the given account and client; the RP will only be given the token, so it'll then probably use it on the UserInfo endpoint to get details about the current user, so pairwise subs can be handled here)

**Comment by jonkoops** [comment]
Hi @tbroyer, we are also leaning towards FedCM and we are interested in possibly implementing this as an experimental feature (see #16834). With Keycloak JS aging and all of the recent changes to 3rd party cookies we're looking for a standardized way to replace it, and FedCM looks like a step forward.

**Reply by tbroyer** [reply]
Ha, didn't find that issue! Maybe I only searched for `fedcm` expecting to at least match the url to the spec 🤷‍♂️

Great news!
